### DISCUSSION TITLE
How should I rewrite the file unicode.rexx of the ARB repository to test it with cReXx?

### DISCUSSION THREAD

**Discussion by jlfaucher** [adesutherland/CREXX#335] [Q&A] [answered]
Hello,
I wanted to test  the output of [ARB unicode.rex](https://github.com/RexxLA/rexx-repository/blob/master/ARB/standards/work-in-progress/code/unicode.rexx) but I realize that it's too early since only levelB is supported.
I'm curious to know how it could be rewritten in levelB. Is there an example that I could study?

Note: the result of the compilation is funny when declaring only
options levelb

```
Error in unicode.rexx @ 9:43 - #SYNTAX_ERROR, "surrogate"
Error in unicode.rexx @ 15:43 - #SYNTAX_ERROR, "grapheme"
2 error(s) in source file

```
I could not believe that the compiler is so hostile against the work in progress in ARB :-)

**Comment by jlfaucher** [comment]
René, I saw your example in the rexxla-arb mailing list to call the BIF length, and tried to build it.
But it appears that calling the BIF length raise a compile error:
```
Importing Procedures - REXX Level A/C/D (cREXX Classic) - Not supported yet
Importing Procedures - INTERNAL ERROR: Compiler Exiting - Failure to create AST
Segmentation fault: 11
```
If I don't call the BIF length then I can build and run the example.
I guess it's because I'm in the branch 'develop'.
Which branch should I use to build your example?

**Comment by rvjansen** [comment]
Hi Jean-Louis,

great that you want to look into it!
I am working from crexx-f0046, this has some scripts that make it easier to work with it - just that part that is missing is filled in by those scripts.
They set the library as relative to the current dir.

I'll try to expand on that in the next email.

best regards,

René.

**Comment by rvjansen** [comment]
I knew I had written that down somewhere: https://github.com/adesutherland/CREXX/discussions/332

this has working scripts for roc including the standard library, rxas and rxvm; a cox script to run and execute, and a crxc to compile a native executable.

best regards,

René.

**Comment by jlfaucher** [comment]
Thanks René
I checked-out  crexx-f0046, removed all the files in the build directory and rebuilt.
Attached: [build.log](https://github.com/adesutherland/CREXX/files/11370291/build.log)

I still have this error
```
Importing Procedures - REXX Level A/C/D (cREXX Classic) - Not supported yet
Importing Procedures - INTERNAL ERROR: Compiler Exiting - Failure to create AST
/Users/Shared/local/rexx/adesutherland/CREXX/build/bin/rxc: line 27: 67362 Segmentation fault: 11  $thisdir/../compiler/rxc -i $thisdir/../lib/rxfns/ ${filename}
```
when compiling with this command `rxc length`
length.rexx contains
```
/* Rexx */
options levelb
import rxfnsb

s1 = 'Café'
s2 = 'Café'

say s1 ':' length(s1)
say s2 ':' length(s2)
```

**Comment by rvjansen** [comment]
Hi Jean-Louis,It should be an out-of-source build; not sure if that is the problem here but I thought I should mention it.Best regards,René.On 2 May 2023, at 08:33, jlfaucher ***@***.***> wrote:﻿
Thanks René
I checked-out  crexx-f0046, removed all the files in the build directory and rebuilt.
Attached: build.log
I still have this error
Importing Procedures - REXX Level A/C/D (cREXX Classic) - Not supported yet
Importing Procedures - INTERNAL ERROR: Compiler Exiting - Failure to create AST
/Users/Shared/local/rexx/adesutherland/CREXX/build/bin/rxc: line 27: 67362 Segmentation fault: 11  $thisdir/../compiler/rxc -i $thisdir/../lib/rxfns/ ${filename}

when compiling with this command rxc length
length.rexx contains
/* Rexx */
options levelb
import rxfnsb

s1 = 'Café'
s2 = 'Café'

say s1 ':' length(s1)
say s2 ':' length(s2)

**Comment by rvjansen** [comment, marked as answer]
And this might help: https://github.com/adesutherland/CREXX/wiki/Building_and_Running_cRexx
There is one bug where it complains about that level A/C/D not supported; caused by scanning other files in the same directory; you can ignore that.
(but not the segfault obv) - conjecture here is that length maybe is not a very lucky name for testing a function of the same name. I will check.

best regards,

René.

**Reply by jlfaucher** [reply]
>There is one bug where it complains about that level A/C/D not supported; caused by scanning other files in the same directory; you can ignore that.

That was the problem! After moving length.rexx in an empty directory, everything worked perfectly.
Thanks!
